src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -151,6 +151,13 @@ def __hash__(self):
     def _parse(self):
         if not sys_info.devices:
             sys_info.devices = disk.get_devices()
+        # check if we are a symlink
+        if os.path.islink(self.abspath):
+            temp_path = os.path.join(os.path.dirname(self.abspath),
+                                            os.readlink(self.abspath))
+            # check if we are not a device mapper
+            if "dm-" not in temp_path:
+                self.abspath = temp_path
```

Comment:
@jansobczak  what was the issue that led to this?

Reply:
As far as I recall this was a case when ceph-volume was used with existing LVMs. Resolving for example `/dev/mapper/vg_name-lv_name -> ../dm-0` which will result in errors when `sys_info.devices` will try to get info.

Reply:
thanks @jansobczak 

I see, I'm not sure if there can be other cases like this one (mpath?)

Reply:
```
# lsblk
NAME     MAJ:MIN RM  SIZE RO TYPE  MOUNTPOINT
sda        8:0    0  200G  0 disk
sdb        8:16   0  200G  0 disk
sdc        8:32   0  200G  0 disk
sdd        8:48   0  200G  0 disk
sde        8:64   0   50G  0 disk
`-mpatha 252:0    0   50G  0 mpath
sdf        8:80   0   50G  0 disk
`-mpatha 252:0    0   50G  0 mpath
sdg        8:96   0   20G  0 disk
`-mpathb 252:1    0   20G  0 mpath
sdh        8:112  0   20G  0 disk
`-mpathb 252:1    0   20G  0 mpath
sdi        8:128  0   20G  0 disk
`-mpathc 252:2    0   20G  0 mpath
sdj        8:144  0   20G  0 disk
`-mpathc 252:2    0   20G  0 mpath
sdk        8:160  0   20G  0 disk
`-mpathd 252:3    0   20G  0 mpath
sdl        8:176  0   20G  0 disk
`-mpathd 252:3    0   20G  0 mpath
sdm        8:192  0   20G  0 disk
`-mpathe 252:4    0   20G  0 mpath
sdn        8:208  0   20G  0 disk
`-mpathe 252:4    0   20G  0 mpath
vda      253:0    0   11G  0 disk
`-vda1   253:1    0   10G  0 part  /
# ls -l /dev/mapper/
total 0
crw-------. 1 root root 10, 236 Jul  6 05:44 control
lrwxrwxrwx. 1 root root       7 Jul  6 05:45 mpatha -> ../dm-0
lrwxrwxrwx. 1 root root       7 Jul  6 05:45 mpathb -> ../dm-1
lrwxrwxrwx. 1 root root       7 Jul  6 05:45 mpathc -> ../dm-2
lrwxrwxrwx. 1 root root       7 Jul  6 05:45 mpathd -> ../dm-3
lrwxrwxrwx. 1 root root       7 Jul  6 05:45 mpathe -> ../dm-4
```

Reply:
@jansobczak I guess we would hit the same issue with mpath devices